lib/include/hse_ikvdb/omf_kmd_vtype.h

```diff
@@ -0,0 +1,24 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+/*
+ * Copyright (C) 2015-2022 Micron Technology, Inc.  All rights reserved.
+ */
+
+#ifndef HSE_KVDB_OMF_KMD_VTYPE_H
+#define HSE_KVDB_OMF_KMD_VTYPE_H
+
+/* Each value has a type indicating how it is stored on media.
+ * Do not change this enum without considering the impact on backward
+ * compatibility.
+ */
+enum kmd_vtype {
+    VTYPE_UCVAL = 0,   // uncompressed value stored in a vblock
+    VTYPE_ZVAL = 1,  // zero-length value (i.e., a key with no value)
+    VTYPE_TOMB = 2,  // tombstone
+    VTYPE_PTOMB = 3, // prefix tombstone
+    VTYPE_IVAL = 4,  // immediate value, uncompressed, stored in a kblock
+    VTYPE_CVAL = 5,  // an LZ4 compressed value stored in a vblock
+};
+
+#define NUM_KMD_VTYPES 6
```

Comment:
VTYPE_MAX ?

Reply:
If `VTYPE_MAX==5`, then it is named accurately but you have to then you have to use `<=` in loops which seems like an invitation for bugs since most loops use `<`.  If `VTYPES_MAX==6` then the loops are natural but the name is wrong.   I see HSE code using both of the MAX approaches: `sp3_work_type` and `sp3_qnum` use the `<MAX` approach,  `cndb_rec_type` uses `<=MAX`.

Reply:
So... I don't like what I wrote, but I like the alternatives less.

Reply:
VTYPE_COUNT would have been a better name and more consistent with other things like HSE_MCLASS_COUNT.